### ISSUE TITLE
Create Local Secondary Index with attribute projection ALL

### ISSUE THREAD

**Issue by ajaishankar** [ServiceStack/PocoDynamo#1]
Currently there is no way to create a local secondary index with an ALL projection.

Looks like only INCLUDE projections are supported and dynamodb enforces the following limit on INCLUDE projections.

http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Limits.html

> You can project a total of up to 20 attributes into all of a table's local and global secondary indexes. This only applies to user-specified projected attributes.

> In a CreateTable operation, if you specify a ProjectionType of INCLUDE, the total count of attributes specified in NonKeyAttributes, summed across all of the secondary indexes, must not exceed 20. If you project the same attribute name into two different indexes, this counts as two distinct attributes when determining the total.

> This limit does not apply for secondary indexes with a ProjectionType of KEYS_ONLY or ALL.

**Comment by mythz** [comment]
This should now be supported [from this commit](https://github.com/ServiceStack/ServiceStack.Aws/commit/e4f66633658076fb502f764aa7e48827e8af5b5b) with the new `[ProjectionType]` attribute, e.g:

```csharp
[ProjectionType(DynamoProjectionType.All)]
public class RockstarAgeAllIndex : IGlobalIndex<Rockstar>
{
    [HashKey]
    public int Age { get; set; }

    [RangeKey]
    public int Id { get; set; }

    public string FirstName { get; set; }
    public string LastName { get; set; }
    public bool Alive { get; set; }
}
```

Or you can use it to annotate a Local Index field, e.g:

```csharp
class Poco
{
   public string HashKey { get; set; }
   public string RangeKey { get; set; }

   [Index]
   [ProjectionType(DynamoProjectionType.All)]
   public string Bar { get; set; }
}
```

This change is available from v4.0.51 that's now [available on MyGet](https://github.com/ServiceStack/ServiceStack/wiki/MyGet).

**Comment by ajaishankar** [comment]
Thanks Demis!
